mod/quiz/accessrule/seb/classes/access_manager.php

```diff
@@ -292,9 +343,50 @@ public function should_validate_config_key() : bool {
      * @return bool
      */
     public function should_validate_browser_exam_key() : bool {
-        return in_array($this->get_seb_use_type(), [
+        $validsebtype = in_array($this->get_seb_use_type(), [
             settings_provider::USE_SEB_UPLOAD_CONFIG,
             settings_provider::USE_SEB_CLIENT_CONFIG,
         ]);
+        return $validsebtype && $this->valid_browser_exam_key_exists();
+    }
+
+    /**
+     * Check that at least one browser exam key exists in the quiz settings.
+     *
+     * @return bool True if one or more keys are set in quiz settings.
+     */
+    public function valid_browser_exam_key_exists(): bool {
+        return !empty($this->quizsettings->get('allowedbrowserexamkeys'));
+    }
+
+    /**
+     * Set session access for quiz.
+     *
+     * @param bool $accessallowed
+     */
+    public function set_session_access(bool $accessallowed): void {
+        global $SESSION;
+        if (!isset($SESSION->quizaccess_seb_access)) {
+            $SESSION->quizaccess_seb_access = [];
+        }
+        $SESSION->quizaccess_seb_access[$this->quiz->get_cmid()] = $accessallowed;
+    }
+
+    /**
+     * Get session access for quiz if already set.
+     *
+     * @return bool
+     */
+    public function get_session_access(): bool {
```

Comment:
should we consistent with all other validate* methods and call it something like "validate_session_access"?

Reply:
Good point, updated.